src/instana/instrumentation/sanic_inst.py

```diff
@@ -1,144 +1,125 @@
-                scope.span.set_tag('span.kind', 'entry')
-                scope.span.set_tag('http.path', request.path)
-                scope.span.set_tag('http.method', request.method)
-                scope.span.set_tag('http.host', request.host)
+                    return
+
+                headers = request.headers.copy()
+                parent_context = tracer.extract(Format.HTTP_HEADERS, headers)
+
+                span = tracer.start_span("asgi", span_context=parent_context)
+                request.ctx.span = span
+
+                ctx = trace.set_span_in_context(span)
+                token = context.attach(ctx)
+                request.ctx.token = token
+
+                span.set_attribute("span.kind", SpanKind.CLIENT)
+                span.set_attribute("http.path", request.path)
+                span.set_attribute(SpanAttributes.HTTP_METHOD, request.method)
+                span.set_attribute(SpanAttributes.HTTP_HOST, request.host)
                 if hasattr(request, "url"):
-                    scope.span.set_tag("http.url", request.url)
+                    span.set_attribute(SpanAttributes.HTTP_URL, request.url)
 
                 query = request.query_string
 
                 if isinstance(query, (str, bytes)) and len(query):
                     if isinstance(query, bytes):
-                        query = query.decode('utf-8')
-                    scrubbed_params = strip_secrets_from_query(query, agent.options.secrets_matcher,
-                                                               agent.options.secrets_list)
-                    scope.span.set_tag("http.params", scrubbed_params)
-
-                if agent.options.extra_http_headers is not None:
-                    extract_custom_headers(scope.span, headers)
-                await wrapped(*args, **kwargs)
+                        query = query.decode("utf-8")
+                    scrubbed_params = strip_secrets_from_query(
+                        query, agent.options.secrets_matcher, agent.options.secrets_list
+                    )
+                    span.set_attribute("http.params", scrubbed_params)
+
+                if agent.options.extra_http_headers:
+                    extract_custom_headers(span, headers)
                 if hasattr(request, "uri_template") and request.uri_template:
-                    scope.span.set_tag("http.path_tpl", request.uri_template)
-                if hasattr(request, "ctx"):  # ctx attribute added in the latest v19 versions
-                    request.ctx.iscope = scope
-        except Exception as e:
-            logger.debug("Sanic framework @ handle_request", exc_info=True)
-            return await wrapped(*args, **kwargs)
-
-
-    logger.debug("Instrumenting Sanic")
+                    span.set_attribute("http.path_tpl", request.uri_template)
+            except Exception:
+                logger.debug("request_with_instana: ", exc_info=True)
```

Comment:
Shouldn't we record exception to the span here?

```suggestion
        except Exception as exc:
                span.record_exception(exc)
                logger.debug("request_with_instana: ", exc_info=True)
```

Reply:
Nope, because this is an exception handled because of the try-except block, not a 500 (or any other status different than the 200 range).